lua/wikis/commons/TeamParticipants/Repository.lua

```diff
@@ -8,52 +8,116 @@
+---@param participant TeamParticipant
+function TeamParticipantsRepository.save(participant)
+	-- Since we merge data from prizepool and teamparticipants, we need to first fetch the existing record from prizepool
+	local lpdbData = TeamParticipantsRepository.getPrizepoolRecordForTeam(participant.opponent) or {}
+
+	local function generateObjectName()
+		local team = Opponent.toName(participant.opponent)
+		local isTbd = Opponent.isTbd(participant.opponent)
+
+		if not isTbd then
+			return 'ranking_' .. mw.ustring.lower(team)
 		end
-	)
-	return Array.map(records, TeamParticipantsRepository.entityFromRecord)
+
+		-- Ensure names are unique for TBDs
+		local storageName = 'participant_' .. mw.ustring.lower(team)
+		local tbdCounter = tonumber(teamCardsVars:get('TBDs')) or 1
+
+		storageName = storageName .. '_' .. tbdCounter
+		teamCardsVars:set('TBDs', tbdCounter + 1)
+
+		return storageName
+	end
+
+	-- Use the tournament defaults if no data is provided from prizepool
+	lpdbData.objectName = lpdbData.objectName or generateObjectName()
+	lpdbData.tournament = lpdbData.tournament or Variables.varDefault('tournament_name')
+	lpdbData.parent = lpdbData.parent or Variables.varDefault('tournament_parent')
+	lpdbData.series = lpdbData.series or Variables.varDefault('tournament_series')
+	lpdbData.shortname = lpdbData.shortname or Variables.varDefault('tournament_tickername')
+	lpdbData.mode = lpdbData.mode or Variables.varDefault('tournament_mode')
+	lpdbData.type = lpdbData.type or Variables.varDefault('tournament_type')
+	lpdbData.liquipediatier = lpdbData.liquipediatier or Variables.varDefault('tournament_liquipediatier')
+	lpdbData.liquipediatiertype = lpdbData.liquipediatiertype or Variables.varDefault('tournament_liquipediatiertype')
+	lpdbData.publishertier = lpdbData.publishertier or Variables.varDefault('tournament_publishertier')
+	lpdbData.icon = lpdbData.icon or Variables.varDefault('tournament_icon')
+	lpdbData.icondark = lpdbData.icondark or Variables.varDefault('tournament_icondark')
+	lpdbData.game = lpdbData.game or Variables.varDefault('tournament_game')
+	lpdbData.startdate = lpdbData.startdate or Variables.varDefault('tournament_startdate')
+	lpdbData.date = lpdbData.date or Variables.varDefault('tournament_enddate')
+
+	lpdbData.qualifier = participant.qualifierText
+	lpdbData.qualifierpage = participant.qualifierPage
+	lpdbData.qualifierurl = participant.qualifierUrl
+
+	lpdbData.extradata = lpdbData.extradata or {}
+	lpdbData.extradata.opponentaliases = participant.aliases
+
+	lpdbData = Table.mergeInto(lpdbData, Opponent.toLpdbStruct(participant.opponent, {setPlayersInTeam = true}))
+	-- Legacy participant fields
+	lpdbData = Table.mergeInto(lpdbData, Opponent.toLegacyParticipantData(participant.opponent))
+	lpdbData.players = lpdbData.opponentplayers
+
+	-- Calculate individual prize money (prize money per player on team)
+	local numberOfPlayersOnTeam = #(participant.opponent.players or {})
```

Comment:
wouldn't we need to filter on the players actually being players and not coaches or something like that?

Reply:
maybe add a TODO comment in this case :)